api/grpcserver/smesher_service.go

```diff
@@ -25,40 +27,34 @@ func (s SmesherService) RegisterService(server *Server) {
 }
 
 // NewSmesherService creates a new grpc service using config data.
-func NewSmesherService(miner api.MiningAPI) *SmesherService {
-	return &SmesherService{
-		Mining: miner,
-	}
+func NewSmesherService(post api.PostAPI, smeshing api.SmeshingAPI) *SmesherService {
+	return &SmesherService{post, smeshing}
 }
 
 // IsSmeshing reports whether the node is smeshing
 func (s SmesherService) IsSmeshing(ctx context.Context, in *empty.Empty) (*pb.IsSmeshingResponse, error) {
 	log.Info("GRPC SmesherService.IsSmeshing")
 
-	stat, _, _, _ := s.Mining.MiningStats()
-	isSmeshing := stat == activation.InitDone
-	return &pb.IsSmeshingResponse{IsSmeshing: isSmeshing}, nil
+	return &pb.IsSmeshingResponse{IsSmeshing: s.smeshing.Smeshing()}, nil
 }
 
 // StartSmeshing requests that the node begin smeshing
 func (s SmesherService) StartSmeshing(ctx context.Context, in *pb.StartSmeshingRequest) (*pb.StartSmeshingResponse, error) {
 	log.Info("GRPC SmesherService.StartSmeshing")
 
+	// TODO(moshababo): remove DataDir and CommitmentSize from the request proto definition.
+	// TODO(moshababo): check why JSON request via HTTP gateway doesn't decode `coinbase` properly
```

Comment:
This is a bit tricky since it's expecting bytes, not a string. Possibly useful: https://github.com/spacemeshos/api/issues/109

Reply:
I've used a similar encoding (and inspected the encoding used in responses), but it doesn't work well for `coinbase` field for some reason (unlike `account_id`, both of the same type). 
We also have examples of this in https://github.com/spacemeshos/cliwallet.

My format:
```
{
    "coinbase": {
        "address": "QWxhZGRpbjpvcGVuIHNlc2FtZQ=="
    }
}
```
I've tried it a while ago. Will give it another shot before merging, just for clients' well-being.

Reply:
That's really strange. `coinbase` in the SmesherService is of type `account_id` which is just an address passed as an array of bytes, as elsewhere in the API. Do you have sample curl commands, one that works somewhere else and one that fails here?

Reply:
It should work ok. CLIWallet uses binary data for all addresses in GRPC calls and it works as expected. There's no need to base64 binary data in grpc - only when working higher-level via the json-proxy as there's no binary data in json.